pkg/crypto/rings/cache.go

```diff
@@ -24,6 +25,13 @@ type ringCache struct {
 	ringPointsCache map[string][]ringtypes.Point
 	ringPointsMu    *sync.RWMutex
 
+	// delegationClient is used to listen for on-chain delegation events and
+	// invalidate cache entries for rings that have been updated on chain.
+	delegationClient client.DelegationClient
+
+	// delegateeChangeObs is the observer that listens for delegatee changes
```

Comment:
:thinking: Maybe something like this?

```suggestion
	// delegateeChangeObs is a replay observable of type DelegateeChange.
	// It is notified of delegate change events as they occur.
```

Reply:
But its an observer? Here I may be a little ocnfused - it is created by subscribing to the delegation client which updates automatically the publisher so this should never need to change - the observer remains constant no?

Reply:
I see, I think I misunderstood previously.

I'm thinking it might be better to hold a reference to the observable instead of the observer. In this case, the main reason would be to mitigate confusion as we consistently use the pattern `<some>Obs` for observables in other places. 

I think, generally, we should prefer holding references to observables rather than observers to simplify future work/refactoring. For example, imagine that we needed to transform the redelegation event stream somehow, or fork the stream. These kinds of changes are trivial if we're holding a reference to the observable but would require additional refactoring (including func/method signature modifications) to support if we only have a reference to an observer.